backend/coins/eth/account.go

```diff
@@ -479,13 +479,17 @@ func (account *Account) Notifier() accounts.Notifier {
 
 // Transactions implements accounts.Interface.
 func (account *Account) Transactions() (accounts.OrderedTransactions, error) {
-	account.Synchronizer.WaitSynchronized()
+	if !account.Synced() || account.Offline() != nil {
+		return nil, accounts.ErrSyncInProgress
+	}
 	return accounts.NewOrderedTransactions(account.transactions), nil
 }
 
 // Balance implements accounts.Interface.
 func (account *Account) Balance() (*accounts.Balance, error) {
-	account.Synchronizer.WaitSynchronized()
+	if !account.Synced() || account.Offline() != nil {
```

Comment:
same, why is Offline() returning ErrSyncInProgress? in fact Offline() returns an error that could be passed along as is?

Reply:
Done